src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -235,13 +242,15 @@ public void close(Collection<TopicPartition> partitions) {
                 tp.partition());
           }
         });
+    unregisterAllSnowflakeJMXMetrics();
```

Comment:
I am not entirely sure if we should handle any exceptions are thrown in the close method before reaching unregisterMetrics. Would it make sense to put the existing code in try { } and the unregister call in a finally { }?

Reply:
Good point, yes. Let me do that. I will wrap it around try-catch-finally

Reply:
Actually, now that I think about this, we don't need to do this in try-catch-finally. We are doing unregistering outside of foreach so it should be fine since we catch all exception inside for each. 

And I will lazily unregister all metrics before we try to register during partition assignment. (Coming in next PR)

I will keep this as is.